dados/Merge_dados/cruzamento.Rmd

```diff
@@ -0,0 +1,49 @@
+---
+title: "Untitled"
+author: "Arthur Sena"
+date: "July 20, 2016"
+output: html_document
+---
+
+```{r}
+
+      ano.atual <- read.csv("~/Documentos/Didatica/ano-atual.csv")
+
+     #Criando um ID para cada deputado
+    deputados <- transform( ano.atual,id=as.numeric(factor(txNomeParlamentar)))
+    
+    #Criando uma tabela com o deputado e seu ID
+    deputados_id <- select(deputados,txNomeParlamentar,id,sgPartido,sgUF) %>% group_by(txNomeParlamentar) %>% slice(c(1))
+
+    #Salvando a tabela anterior
+    write.table(deputados_id,file = "deputados_id.csv", sep = "," ,row.names = F)
+    
+    #Tabela se fazer o merge
+    siconv_emenda <- read.csv("siconv_emenda.csv", sep=";")
+    
+    #Usando semi join do dplyr. Ele mantém apenas os deputados que conseguiu fazer o match
+    colnames(deputados)[5] = "NOME_PARLAMENTAR"
+    l <- deputados %>% semi_join(siconv_emenda) 
+    length(unique(l$id)) #Ele fica com mais ou menos 386 deputados :/
```

Comment:
Certo, mas no caso então deixa a versão mais precisa pronta e documenta o
que é pra ser executado e que resultado dá por favor

On Thu, Jul 21, 2016 at 7:28 PM arthurSena <notifications@github.com> wrote:

> In dados/Merge_dados/cruzamento.Rmd
> <https://github.com/nazareno/gastos-da-camara/pull/1#discussion_r71799096>
> :
>
> > +     #Criando um ID para cada deputado
> > +    deputados <- transform( ano.atual,id=as.numeric(factor(txNomeParlamentar)))
> > +
> > +    #Criando uma tabela com o deputado e seu ID
> > +    deputados_id <- select(deputados,txNomeParlamentar,id,sgPartido,sgUF) %>% group_by(txNomeParlamentar) %>% slice(c(1))
> > +
> > +    #Salvando a tabela anterior
> > +    write.table(deputados_id,file = "deputados_id.csv", sep = "," ,row.names = F)
> > +
> > +    #Tabela se fazer o merge
> > +    siconv_emenda <- read.csv("siconv_emenda.csv", sep=";")
> > +
> > +    #Usando semi join do dplyr. Ele mantém apenas os deputados que conseguiu fazer o match
> > +    colnames(deputados)[5] = "NOME_PARLAMENTAR"
> > +    l <- deputados %>% semi_join(siconv_emenda)
> > +    length(unique(l$id)) #Ele fica com mais ou menos 386 deputados :/
>
> 386 ? Sim ainda é.
> O join do dplyr só faz o match se os nomes forem exatamente iguais.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/nazareno/gastos-da-camara/pull/1/files/8b80698583e3d2df9972f89da00b970a96349dcc#r71799096>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABAUmtkG9NtWSiv1BO93ozTujE1jg2JTks5qX_JzgaJpZM4JSQzb>
> .
>